FIX: Move ``_stim.<ext>`` specification within the Task Events module (#1750)
bids-standard/bids-specification: src/modality-specific-files/physiological-recordings.md

For example for the file `sub-control01_task-nback_run-1_bold.nii.gz`,
`<matches>` would correspond to `sub-control01_task-nback_run-1`.

> !!! warning "TSVGZ files MUST NOT include a header line (as established by the [common-principles](../common-principles.md#compressed-tabular-files))"
> 
>     As a result, when supplying a `*_physio.tsv.gz` file, an accompanying
>     `*_physio.json` MUST be supplied as well.

Author: ```suggestion
!!! warning "Caution"

    TSVGZ files MUST NOT include a header line (as established by the
    [common-principles](../common-principles.md#compressed-tabular-files))

    As a result, when supplying a `*_physio.tsv.gz` file, an accompanying
    `*_physio.json` MUST be supplied as well.
```

`sub-01_task-bart_recording-breathing_physio.tsv.gz` to contain respiratory
measurements in a different sampling frequency.

> Physiological recordings (including eyetracking) SHOULD use the `_physio`.

Reviewer: ```suggestion
Physiological recordings (including eyetracking) SHOULD use the `_physio` suffix.
```

`sub-01_task-bart_recording-breathing_physio.tsv.gz` to contain respiratory
measurements in a different sampling frequency.

> Physiological recordings (including eyetracking) SHOULD use the `_physio` suffix.

Reviewer 2: Shouldn't that be a MUST and not a SHOULD?

> Physiological recordings (including eyetracking) SHOULD use the `_physio` suffix.

Reviewer 2: ```suggestion
Physiological recordings (including eyetracking) MUST use the `_physio` suffix.
```

> The following table specifies metadata fields for the `*_physio.json` file.

Reviewer 2: Actually because we've just split things into several tables, I think we should keep the `tables specify`, no?

Sorry about this.

> The following table specifies metadata fields for the `*_physio.json` file.

Reviewer 2: ```suggestion
The following tables specifiy metadata fields for the `*_physio.json` file.
```

(and the [`recording-<label>`](../appendices/entities.md#recording)
entity MAY be used to distinguish these files).

If the same continuous recording has been used for all subjects (for example in
the case where they all watched the same movie), one file MAY be used and
placed in the root directory.
For example, `task-movie_stim.tsv.gz`

> 

Reviewer: Should we note something like

> For simultaneous recordings, such as fMRI and EEG,
> physiological recordings MAY be placed as a single file in the subject or session directory.

Or should we leave well enough alone?

Author: A note on that makes sense, but I think it deserves opening a new issue.

While it is not at all weird to set on your physiological recording gear at the beginning of the session and switch it off only at the end, cropping and splitting these files is usually a difficult task. I'm afraid that stuffing a single file could turn the recordings useless if the end-user does not have sufficient information to cut them at the right places (e.g., they come without triggers or flags). This is made worse by the fact that BIDS is quite sparse in explicit metadata recommendations to ensure the splitting of these files is feasible.

cc/ @smoia who will be extremely interested in this.

Author: That said, if the recordings would  be labeled with a column "experiment time" given by the scanner and in sync with the `scans.tsv` and start of each run, this option would definitely be the best option (with some caveats such as dummy scans if they are not properly marked).

Reviewer 3: I have the feeling that @effigies and @oesteban you're raising awareness on two different points - I'll comment both what I interpreted @effigies to say and what @oesteban I think you're referring to.

@effigies if by simultaneous recordings you mean cases like concurrent EEG/fMRI setups or EEG/motion cameras, that would normally create two different set of data having their own matching physio files, then I guess your issue is not storing the physio files twice when it's clearly all coming from one run.
IMHO this comes from (not being able to) defining "dependencies" or hierarchy between different files/modalities, and it inversely relates to another issue we're facing in physiopy, that is where to store "redundant" physio data that are collected before/after a "matching neuro run" (currently phys2bids approaches this as leaving it in the first run, but we're thinking about splitting them and moving them to a `beh` folder without a clear "matching dependency" for lack of currently better options).
Going back to your point, how would you declare that those physio files can be used in either modality? Would it be better to have a subject or session folder with 3 to 10 extra files, or to consider instead what equipment "created" those files (e.g. MRI vs external physio equipment vs EEG equipment)? Alternatively, could we consider having a `physio` folder for these types of cases?

@oesteban I think you're referring more to the multiple-neuro-runs-in-a-physio-recording issue, where we normally leave our physio equipment recording for the longest time and we then have to find a solution to split it to match it with neuro data. In that case, besides the fact that it's not compatible with the current specification, I agree with you that dumping a the single longest timeseries ever might confuse users (although that is the best option to allow extra recording for lag-related analyses).

Author: I think you are right; I misinterpreted @effigies' message.

Author: And I made another "mistake" in that I thought about physio, when here we are talking about `_stim.tsv.gz`. That also pushed your comment away.

- For `_physio` this is worth opening an issue (cc @Remi-Gau) because whatever we do (`beh/`, or `physio/`, or single file at session level) requires some thought.
- For `_stim` this is relatively easier because it feels this should go into `beh/` if a `stim` file responds to fMRI and EEG recorded simultaneously.

WDYT?

Reviewer 3: (@isesteves @me-pic @mmiedema if you want to follow the convo)

Author: Please redirect the physio part of this problem to #1753. I'll fix this in the mean time for `stim` and point at `beh`.

In the case of `stim`, you can definitely run the same movie (please read stimulus) with fMRI and EEG separately, so this is actually more likely to happen in standard experiments.

(and the [`recording-<label>`](../appendices/entities.md#recording)
entity MAY be used to distinguish these files).

For motion parameters acquired from MRI scanner side motion correction, the

> `_physio` suffix SHOULD be used.

Reviewer 2: unrelated but this `SHOULD` should be a `MUST` too, no?

> `_physio` suffix SHOULD be used.

Reviewer 2: ```suggestion
`_physio` suffix MUST be used.
```
